mongo/client_encryption.go

```diff
@@ -72,6 +73,62 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	return ce, nil
 }
 
+// CreateEncryptedCollection creates a new collection with a help of automatic generation of new encryption data keys for null keyIds.
+// The EncryptedFields field in createOpts will be updated with new keys.
+func (ce *ClientEncryption) CreateEncryptedCollection(ctx context.Context,
+	db *Database, coll string, createOpts *options.CreateCollectionOptions,
+	kmsProvider string, dkOpts *options.DataKeyOptions) (*Collection, error) {
+	if createOpts == nil {
```

Comment:
Should we error here? Can you not create an encrypted collection with nil `createOpts` and rely on the `EncryptedFields` from the `EncryptedFieldsMap`?

Reply:
Found the spec [here](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#collection-encryptedfields-lookup-getencryptedfields). I think it is a proper fallback.

Reply:
I also see that the C# implementation ensures the nonnullness of the `createOpts`; seems fine to keep this here.